src/test/java/core/basesyntax/service/impl/FruitShopServiceImplTest.java

```diff
@@ -0,0 +1,105 @@
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.FruitShopService;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.OperationStrategyImpl;
+import core.basesyntax.strategy.handlerimpl.BalanceOperation;
+import core.basesyntax.strategy.handlerimpl.PurchaseOperation;
+import core.basesyntax.strategy.handlerimpl.ReturnOperation;
+import core.basesyntax.strategy.handlerimpl.SupplyOperation;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.jupiter.api.Assertions;
+
+public class FruitShopServiceImplTest {
+    public static final Integer BANANA_AMOUNT = 152;
+    public static final Integer APPLE_AMOUNT = 90;
+    private static FruitShopService fruitShopService;
+    private static List<FruitTransaction> fruitTransactions;
+    private static Map<String,Integer> expectedFruitStorage;
+
+    @BeforeClass
+    public static void setUp() {
+        Map<FruitTransaction.Operation, OperationHandler> operationHandlerMap = Map.of(
+                FruitTransaction.Operation.BALANCE, new BalanceOperation(),
+                FruitTransaction.Operation.PURCHASE, new PurchaseOperation(),
+                FruitTransaction.Operation.SUPPLY, new SupplyOperation(),
+                FruitTransaction.Operation.RETURN, new ReturnOperation()
+        );
+        fruitShopService = new FruitShopServiceImpl(new OperationStrategyImpl(operationHandlerMap));
+        expectedFruitStorage = new HashMap<>();
+        fruitTransactions = new ArrayList<>();
+    }
+
+    @Before
+    public void init() {
+        fruitTransactions = Stream.of(
+                        new FruitTransaction(FruitTransaction.Operation.BALANCE,
+                                "banana", 20),
+                        new FruitTransaction(FruitTransaction.Operation.SUPPLY,
+                                "apple", 100),
+                        new FruitTransaction(FruitTransaction.Operation.SUPPLY,
+                                "banana", 100),
+                        new FruitTransaction(FruitTransaction.Operation.PURCHASE,
+                                "banana", 13),
+                        new FruitTransaction(FruitTransaction.Operation.RETURN,
+                                "apple", 10),
+                        new FruitTransaction(FruitTransaction.Operation.PURCHASE,
+                                "apple", 20),
+                        new FruitTransaction(FruitTransaction.Operation.PURCHASE,
+                                "banana", 5),
+                        new FruitTransaction(FruitTransaction.Operation.SUPPLY,
+                                "banana", 50))
+                .collect(Collectors.toList());
+    }
+
+    @Test
+    public void processData_ok() {
```

Comment:
also let's rename your test methods following this convention: `<methodUnderTest>_<state>_<expectedBehavior>`